Log stack traces for PHP warnings etc

katebutler opened the pull request:
Includes the full stack in the log message when writing to file only.  The log message for other handlers (web notifications and database) will not be changed.

Commit:
Allow stack traces to be logged to file for PHP warnings

Commit:
Use debug_backtrace() to generate an abbreviated stack trace

tsteur commented:
BTW @katebutler there are some failing tests https://travis-ci.org/matomo-org/matomo/jobs/594801494 and https://travis-ci.org/matomo-org/matomo/jobs/594801495#L818

and as discussed be good to remove the newline.

Commit:
Change trace format, fix LogTest

tsteur commented:
@katebutler I'm thinking this failing test might be due to this change? https://travis-ci.org/matomo-org/matomo/jobs/594907204#L817

tsteur commented on core/ErrorHandler.php:
```diff
@@ -161,8 +161,12 @@ public static function errorHandler($errno, $errstr, $errfile, $errline)
             case E_DEPRECATED:
             case E_USER_DEPRECATED:
             default:
+                $context = array('trace' => debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS));
```
@katebutler could we add a limit here of say max 15 last traces? There is a second parameter `$limit` that can be set.

tsteur committed:
smaller trace by not including unimportant doc root 

makes it also easier for users to share the trace since they don't need to remove any paths

tsteur committed:
Update ErrorHandler.php
